src/compute/src/sink/persist_sink.rs

```diff
@@ -1181,3 +1191,87 @@ where
     let token = Rc::new(shutdown_button.press_on_drop());
     (output_stream, token)
 }
+
+/// This is for REFRESH options on materialized views. It adds an operator that rounds up the
+/// timestamps of data and frontiers to the time of the next refresh. See
+/// `doc/developer/design/20231027_refresh_every_mv.md`.
+///
+/// Note that this currently only works with 1-dim timestamps. (This is not an issue for WMR,
+/// because iteration numbers should disappear by the time the data gets to the Persist sink.)
+fn round_up<G>(
+    coll: Collection<G, Result<Row, DataflowError>, Diff>,
+    refresh_schedule: RefreshSchedule,
+) -> Collection<G, Result<Row, DataflowError>, Diff>
+where
+    G: Scope<Timestamp = Timestamp>,
+{
+    let mut builder = OperatorBuilder::new("round_up".to_string(), coll.scope().clone());
+    let (mut output_buf, output_stream) = builder.new_output();
+    let mut input = builder.new_input_connection(&coll.inner, Pipeline, vec![Antichain::new()]);
+    builder.build(move |capabilities| {
+        let mut capability = Some(capabilities.into_element());
```

Comment:
Maybe also add a comment here to explain that:
* This capability directly controls this operators output frontier (because we have disconnected the input above).
* We wrap it in an `Option` so we can drop it to advance to the empty output frontier when we know no refresh will happen again.
* We must be careful that we only ever emit output updates at times that are at or beyond this capability.

Reply:
Done